src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -794,16 +830,16 @@ private VarSymbol makeSyntheticVar(long flags, Name name, Type type, Symbol owne
     private MethodType typeToMethodType(Type mt) {
         Type type = types.erasure(mt);
```

Comment:
not part of your patch but I think that `mt` should already be erased at this point, now that Lower is running before L2M

Reply:
I don't think so - I see it used here:

```
MethodSymbol samSym = (MethodSymbol) types.findDescriptorSymbol(tree.target.tsym);
List<LoadableConstant> staticArgs = List.of(
                typeToMethodType(samSym.type),
                refSym.asHandle(),
                typeToMethodType(tree.getDescriptorType(types)));
```

I think neither `samSym.type` nor the result of `tree::getDescriptorType` (which is based on `tree.target`) are erased.